src/Microsoft.XmlSerializer.Generator/src/Microsoft/XmlSerializer/Generator/XmlSerializationReader.cs

```diff
@@ -2197,6 +2200,8 @@ private void WriteMemberElementsIf(Member[] members, Member anyElement, string e
                 Writer.WriteLine("switch (state) {");
             }
             int cases = 0;
+            bool largeNumberOfMembers = (members.Length > 1000);
```

Comment:
Why is the number set to 1000? 

Besides, I think it'd be better to exact the value into a readonly variable.

Reply:
Unless there are perf concerns, I would not bother to generate different code.